FHIRClientDevelopmentGuide.md

```diff
@@ -0,0 +1,100 @@
+# FHIR Client Development Guide
+November 17, 2022
+
+[Introduction](#introduction)  
+[Integration with EDRS System](#integration-with-edrs-system)  
+[Authentication](#authentication)  
+[Timeliness](#timeliness)  
+[Message Resends](#message-resends)  
+[Response Message Handling](#response-message-handling)  
+[Message Traceabality](#message-traceabality)  
+[NCHS Backup Endpoint](#nchs-backup-endpoint)  
+
+## Introduction
+The NVSS FHIR Modernization effort will implement the exchange of FHIR messages between jurisdiction’s and NCHS for mortality data. NCHS has implemented a test FHIR API server for jurisdictions to submit their mortality data. Each jurisdiction will implement a client that exchanges FHIR data with the NCHS server. A jurisdiction’s client implementation must follow the messaging protocol described in the [FHIR Messaging IG](http://build.fhir.org/ig/nightingaleproject/vital_records_fhir_messaging_ig/branches/main/message.html) to ensure reliable message delivery and traceability. The [Reference Client](https://github.com/nightingaleproject/Reference-Client-API) is an open-source example of a C# implementation of the FHIR Messaging IG. Implementers may leverage the Reference Client, tools such as Rhapsody, or develop a custom solution to communicate with the API. In general, each Client implementation should support the following guidelines when deployed to production. Each section species configurable parameters that should be defined in a separate config file.
+
+## Integration with EDRS System
+The client must be integrated with the jurisdiction’s EDRS system so that FHIR Death Records produced by the EDRS system are submitted to the FHIR API by the client.
+
+The client should be capable of submitting the following message types to the API:
+- Submission Message
+- Update Message
+- Alias Message
+- Void Message
+- Acknowledgements
+
+## Authentication 
+The client must support authentication to the FHIR API gateway
+- Jurisdictions must have STEVE credentials. STEVE credentials must be used by the client to make authenticated requests to STEVE to submit data to the FHIR API on behalf of the jurisdiction
+- Jurisdictions must have SAMS credentials. SAMS credentials must be used by the client to make authenticated requests to the backup API endpoint
+The client should support automatic authentication refresh
+- The STEVE auth token expires after 5 minutes. Client systems should be capable of auto refreshing their token for uninterrupted communications. 
+- The SAMS auth token expires after 1 hour. Client systems should be capable of auto refreshing their token for uninterrupted communications. 
+
+### Configurable Parameters
+- STEVE Username
+- STEVE Password
+- STEVE Client Secret
+- STEVE Client Token
+- STEVE Authentication endpoint
+- SAMS Username
+- SAMS Password
+- SAMS Client Secret
+- SAMS Client Token
+- SAMS Authentication endpoint
```

Comment:
Maybe the API endpoints should be configurable? And also which endpoint is currently selected?